### PR TITLE
NEXT-16225 - Added custom entities documentation

### PR DESCRIPTION
**Pull request by OliverSkroblin**

### CODE REVIEW

File: guides/plugins/apps/custom-data/custom-entities.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Custom entities
+
+You can add custom entities to Shopware and thus extend the functionality of your shop.
```
Comment (keulinho) [inline guides/plugins/apps/custom-data/custom-entities.md:3]: can you add a little bit more explanatory text on where to use the feature?

E.g. if you want to store custom complex data in the shopware DB you can use custom entities to do so...

File: guides/plugins/apps/custom-data/custom-entities.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Custom entities
+
+In addition to [Custom fields](custom-fields.md), you can create completely own entities in the system, named custom entities.
+Unlike [Custom fields](custom-fields.md), you can generate completely custom data structures with custom relations, which can then be maintained by the admin.
+To make use of the custom entities register your entities in your `entities.xml` file, which is located in the `config` directory of your app.
+
+{% code title="entities.xml" %}
+```markup
```
Comment (keulinho) [inline guides/plugins/apps/custom-data/custom-entities.md:8]:
```suggestion
```xml
```
code higlighting will be better

File: guides/plugins/apps/custom-data/custom-entities.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            <many-to-many name="products" reference="product" store-api-aware="true" />
+        </fields>
+    </entity>
+</entities>
+```
+{% endcode %}
+
+For a complete reference of the structure of the entities file take a look at the [Custom entity xml reference](../../../resources/references/app-reference/entities-reference.md).
+
+## Functionality
```
Comment (JoshuaBehrens) [inline guides/plugins/apps/custom-data/custom-entities.md:24]: New lines after headlines is nice

**Review by JoshuaBehrens — CHANGES_REQUESTED** [review]

File: guides/plugins/apps/custom-data/custom-entities.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+{% set blogs = services.repository.search('custom_entity_blog', criteria) %}
+```
+{% endraw %}
+
+Additionally, to the repository you can also access your custom entities via [Admin api](../../../../concepts/api/admin-api.md).
+```bash
+POST /api/search/custom-entity-blog
+```
+
+## Permissions
```
Comment (OliverSkroblin) [inline guides/plugins/apps/custom-data/custom-entities.md:37]:
```suggestion
## Permissions

```

File: guides/plugins/apps/custom-data/custom-entities.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            <many-to-many name="products" reference="product" store-api-aware="true" />
+        </fields>
+    </entity>
+</entities>
+```
+{% endcode %}
+
+For a complete reference of the structure of the entities file take a look at the [Custom entity xml reference](../../../resources/references/app-reference/entities-reference.md).
+
+## Functionality
```
Comment (OliverSkroblin) [inline guides/plugins/apps/custom-data/custom-entities.md:24]:
```suggestion
## Functionality

```

**Review by elkmod — APPROVED** [review]

**Comment by JoshuaBehrens** [comment]
Maybe I missed it, but I am missing a hint. I see references from the blog the product. Are they added automatically from product to blog as extension as well? This way you can also query this relation via API. If this already happens it would be cool if it is mentioned.

**Review by elkmod — APPROVED** [review]

**Pull request merged**